.gitignore

```diff
@@ -1,4 +1,5 @@
 target
 .ninja_deps
 .ninja_log
-
+Cargo.lock
```

Comment:
I added these because the Cargo lockfiles were generated by rust-analyzer for me, and I didn't want to check them in accidentally. We should make a decision on lockfiles before merging.

Reply:
I think lockfiles for binaries are appropriate, especially because we might be depending on non-released versions of libs on and off.